src/main.rs

```diff
@@ -0,0 +1,574 @@
+use rand::{thread_rng, Rng};
+use std::collections::HashMap;
+use std::fmt;
+use std::fs;
+use std::fs::File;
+use std::io;
+use std::io::Write;
+use std::ops::Deref;
+use std::path::Path;
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, SystemTime};
+use tokio::sync::mpsc;
+
+#[derive(PartialEq)]
+pub(crate) enum HTLCDirection {
+	Inbound,
+	Outbound,
+}
+
+pub(crate) enum HTLCStatus {
+	Pending,
+	Succeeded,
+	Failed,
+}
+
+pub(crate) struct SatoshiAmount(Option<u64>);
+
+impl fmt::Display for SatoshiAmount {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		match self.0 {
+			Some(amt) => write!(f, "{}", amt),
+			None => write!(f, "unknown"),
+		}
+	}
+}
+
+pub(crate) type PaymentInfoStorage = Arc<
+	Mutex<
+		HashMap<PaymentHash, (Option<PaymentPreimage>, HTLCDirection, HTLCStatus, SatoshiAmount)>,
+	>,
+>;
+
+type ChainMonitor = chainmonitor::ChainMonitor<
+	InMemorySigner,
+	Arc<dyn Filter>,
+	Arc<BitcoindClient>,
+	Arc<BitcoindClient>,
+	Arc<FilesystemLogger>,
+	Arc<FilesystemPersister>,
+>;
+
+pub(crate) type PeerManager = SimpleArcPeerManager<
+	SocketDescriptor,
+	ChainMonitor,
+	BitcoindClient,
+	BitcoindClient,
+	dyn chain::Access,
+	FilesystemLogger,
+>;
+
+pub(crate) type ChannelManager =
+	SimpleArcChannelManager<ChainMonitor, BitcoindClient, BitcoindClient, FilesystemLogger>;
+
+async fn handle_ldk_events(
+	peer_manager: Arc<PeerManager>, channel_manager: Arc<ChannelManager>,
+	chain_monitor: Arc<ChainMonitor>, bitcoind_client: Arc<BitcoindClient>,
+	keys_manager: Arc<KeysManager>, payment_storage: PaymentInfoStorage, network: Network,
+) {
+	let mut pending_txs: HashMap<OutPoint, Transaction> = HashMap::new();
+	loop {
+		peer_manager.process_events();
+		let loop_channel_manager = channel_manager.clone();
+		let mut events = channel_manager.get_and_clear_pending_events();
+		events.append(&mut chain_monitor.get_and_clear_pending_events());
+		for event in events {
+			match event {
+				Event::FundingGenerationReady {
+					temporary_channel_id,
+					channel_value_satoshis,
+					output_script,
+					..
+				} => {
+					// Construct the raw transaction with one output, that is paid the amount of the
+					// channel.
+					let addr = WitnessProgram::from_scriptpubkey(
+						&output_script[..],
+						match network {
+							Network::Bitcoin => bitcoin_bech32::constants::Network::Bitcoin,
+							Network::Testnet => bitcoin_bech32::constants::Network::Testnet,
+							Network::Regtest => bitcoin_bech32::constants::Network::Regtest,
+							Network::Signet => panic!("Signet unsupported"),
+						},
+					)
+					.expect("Lightning funding tx should always be to a SegWit output")
+					.to_address();
+					let mut outputs = vec![HashMap::with_capacity(1)];
+					outputs[0].insert(addr, channel_value_satoshis as f64 / 100_000_000.0);
+					let raw_tx = bitcoind_client.create_raw_transaction(outputs).await;
+
+					// Have your wallet put the inputs into the transaction such that the output is
```

Comment:
Right, I guess we can drop half the code handling this stuff when we land https://github.com/rust-bitcoin/rust-lightning/pull/856

Reply:
It was merged :tada: